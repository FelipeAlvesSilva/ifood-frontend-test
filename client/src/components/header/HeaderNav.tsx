import React from 'react';
import { Link } from 'react-router-dom';
import SpotifyLogo from '../../images/Spotify_Icon_CMYK_Green.png'
import { FaSun as Sun } from 'react-icons/fa';
import { IoMdMoon as Moon } from 'react-icons/io';
import { Switch } from 'antd';
import { Title } from '../styled/Title';

type HeaderProps = {
  toggle: Function
};

const Header = (props: HeaderProps) => (
  <div className="container my-4">
    <Title className="m-5">
      <img className="mr-3" src={SpotifyLogo} alt="Spotify brand icon" style={{ height: '60px', width: '60px' }} />
      <Link to="/home">
        Spotifood
      </Link>
      <Switch
        onClick={() => props.toggle()}
        style={{ float: 'right' }}
        checkedChildren={<Moon className="switcher-icon" />}
        unCheckedChildren={<Sun className="switcher-icon" />}
        defaultChecked
      />
    </Title>
  </div>
);

export default Header;