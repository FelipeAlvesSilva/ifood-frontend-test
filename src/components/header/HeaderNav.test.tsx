import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import HeaderNav from '../../components/header/HeaderNav';

describe('Tests for Header component', () => {
  test('Component should not regress', () => {
    const container = render(
      <Router>
        <HeaderNav toggle={jest.fn()} />
      </Router>);
    expect(container).toMatchSnapshot();
  });
});