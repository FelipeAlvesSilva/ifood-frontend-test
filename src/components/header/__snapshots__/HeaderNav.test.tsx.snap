// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Header component Component should not regress 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container my-4"
      >
        <h1
          class="sc-AxjAm kdfdMB m-5"
        >
          <img
            alt="Spotify brand icon"
            class="mr-3"
            src="Spotify_Icon_CMYK_Green.png"
            style="height: 60px; width: 60px;"
          />
          <a
            class="custom-link"
            href="/home"
          >
            Spotifood
          </a>
          <button
            aria-checked="true"
            class="ant-switch ant-switch-checked"
            role="switch"
            style="float: right;"
            type="button"
          >
            <div
              class="ant-switch-handle"
            />
            <span
              class="ant-switch-inner"
            >
              <svg
                class="switcher-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M195 125c0-26.3 5.3-51.3 14.9-74.1C118.7 73 51 155.1 51 253c0 114.8 93.2 208 208 208 97.9 0 180-67.7 202.1-158.9-22.8 9.6-47.9 14.9-74.1 14.9-106 0-192-86-192-192z"
                />
              </svg>
            </span>
          </button>
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container my-4"
    >
      <h1
        class="sc-AxjAm kdfdMB m-5"
      >
        <img
          alt="Spotify brand icon"
          class="mr-3"
          src="Spotify_Icon_CMYK_Green.png"
          style="height: 60px; width: 60px;"
        />
        <a
          class="custom-link"
          href="/home"
        >
          Spotifood
        </a>
        <button
          aria-checked="true"
          class="ant-switch ant-switch-checked"
          role="switch"
          style="float: right;"
          type="button"
        >
          <div
            class="ant-switch-handle"
          />
          <span
            class="ant-switch-inner"
          >
            <svg
              class="switcher-icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M195 125c0-26.3 5.3-51.3 14.9-74.1C118.7 73 51 155.1 51 253c0 114.8 93.2 208 208 208 97.9 0 180-67.7 202.1-158.9-22.8 9.6-47.9 14.9-74.1 14.9-106 0-192-86-192-192z"
              />
            </svg>
          </span>
        </button>
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
